src/test/java/edu/aau/groupc/canteenbackend/user/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,37 @@
+package edu.aau.groupc.canteenbackend.user.controllers;
+
+import edu.aau.groupc.canteenbackend.endpoints.AbstractControllerTest;
+import edu.aau.groupc.canteenbackend.user.User;
+import edu.aau.groupc.canteenbackend.user.services.IUserService;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("H2Database")
+class UserControllerTest extends AbstractControllerTest {
```

Comment:
You can add `@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)` here to make each test method run on a fresh database. This avoids weird test sequence issues.